src/Microsoft.AspNetCore.Mvc.Core/ProblemErrorPolicyAttribute.cs

```diff
@@ -0,0 +1,132 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.ApiExplorer;
+using Microsoft.AspNetCore.Mvc.Filters;
+
+namespace Microsoft.AspNetCore.Mvc
+{
+    [AttributeUsage(AttributeTargets.Method | AttributeTargets.Class, AllowMultiple = false, Inherited = true)]
+    public class ProblemErrorPolicyAttribute : Attribute, IErrorPolicy, IExceptionFilter, IResultFilter
+    {
+        public void Apply(ErrorPolicyContext context)
+        {
+            var parameters = context.Description.ActionDescriptor.Parameters;
+            if (parameters.Count > 0)
+            {
+                context.Description.SupportedResponseTypes.Add(CreateProblemResponse(StatusCodes.Status400BadRequest));
+
+                if (parameters.Any(p => string.Equals("id", p.Name, StringComparison.OrdinalIgnoreCase)))
+                {
+                    context.Description.SupportedResponseTypes.Add(CreateProblemResponse(StatusCodes.Status404NotFound));
+                }
+            }
+
+            context.Description.SupportedResponseTypes.Add(CreateProblemResponse(statusCode: 0));
+
+            ApiResponseType CreateProblemResponse(int statusCode)
+            {
+                return new ApiResponseType
+                {
+                    ApiResponseFormats = new List<ApiResponseFormat>
+                    {
+                        new ApiResponseFormat
+                        {
+                            MediaType = "application/problem+json",
+                        },
+                    },
+                    ModelMetadata = context.MetadataProvider.GetMetadataForType(typeof(Problem)),
+                    StatusCode = statusCode,
+                    Type = typeof(Problem),
+                };
+            }
+        }
+
+        public void OnException(ExceptionContext context)
+        {
+            if (context.Result == null)
+            {
+                var problem = new Problem
+                {
+                    Type = context.Exception.HelpLink,
+                    Title = context.Exception.Message,
```

Comment:
I guess we could use a standard "An unknown error has occurred." That said, "SHOULD" is somewhat more lenient than "MUST", so maybe this is fine as it is.